piAIOComm.c

```diff
@@ -318,7 +318,7 @@ u32 piAIOComm_sendCyclicTelegram(u8 devnum)
 
 	ret = pibridge_req_io(addr, IOP_TYP1_CMD_DATA, snd_buf,
 			      AIO_OUTPUT_DATA_LEN, rcv_buf, AIO_INPUT_DATA_LEN);
-	if (ret) {
+	if (ret < 0) {
```

Comment:
I think you need to check for `ret != AIO_INPUT_DATA_LEN` instead so that you treat too-short packets as an error (as before).